pkg/extract/extract.go

```diff
@@ -139,10 +139,6 @@ func unzip(archive, target string, fileFilter func(string) bool) ([]string, erro
 			return nil, fmt.Errorf("%s: illegal file path", path)
 		}
 
-		if fileFilter != nil && !fileFilter(path) {
-			logging.Debugf("untar: Skipping %s", path)
-			continue
-		}
```

Comment:
The current code is
```
// the target location where the dir/file should be created
		path := filepath.Join(targetDir, header.Name)
		if fileFilter != nil && !fileFilter(path) {
			logging.Debugf("untar: Skipping %s", path)
			continue
		}

		// check the file type
		switch header.Typeflag {
		// if it's a file create it
		case tar.TypeReg, tar.TypeGNUSparse:
			// tar.Next() will externally only iterate files, so we might have to create intermediate directories here
			if err = os.MkdirAll(filepath.Dir(path), 0750); err != nil {
				return err
			}
			file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, header.FileInfo().Mode())
			if err != nil {
				return err
			}
			defer file.Close()

			// copy over contents
			if _, err := io.Copy(file, tarReader); err != nil {
				return err
			}
		}
```
There is a call to `mkdirAll(path)` before trying to create the file from the archive, is this not working?

Reply:
@cfergeau no this not going to work, because if you check the current code the `path` here is got from `path := filepath.Join(targetDir, header.Name)` and `header.Name` only contain the file name not the complete path so in our current scenario we check first the `fileFilter(path)` and skip that directory creation if the binary is inside a sub-directory. 

Assume binary name is `oc` which we want to extract and it is available in the tarball like `A/B/oc` so as per current flow when tar.Next() hit the `B` then fileFilter check if the `B` is `oc` if not then continue without creating the directory `B` and then when tar.Next() counter to `oc` it identify the binary is `oc` and now it hit the ` os.MkdirAll(filepath.Dir(path), 0750)` where the `path` will only have `<target_dir>/oc` instead `<target_dir>/B/oc` and it will fail.